### PR TITLE
add pass profiler

### PR DESCRIPTION
**Pull request by rkimballn1**
To enable the pass profiler set the environment varialble `NGRAPH_PROFILE_PASS_ENABLE`

### CODE REVIEW

File: src/ngraph/pass/manager.cpp [hunk 1]
```diff
@@ -122,6 +131,19 @@ void ngraph::pass::Manager::run_passes(shared_ptr<Function> func)
             }
         }
         index++;
+        pass_timer.stop();
+        if (profile_enabled)
+        {
+            PassBase* p = pass.get();
+            string name = typeid(*p).name();
+            int status;
+            name = abi::__cxa_demangle(name.c_str(), 0, 0, &status);
```
Comment (Krovatkin) [inline src/ngraph/pass/manager.cpp]: This is so awesome!! I didn't realize we can call a demangler programmatically.

File: src/ngraph/pass/manager.cpp [hunk 2]
```diff
@@ -122,6 +131,19 @@ void ngraph::pass::Manager::run_passes(shared_ptr<Function> func)
             }
         }
         index++;
+        pass_timer.stop();
+        if (profile_enabled)
+        {
+            PassBase* p = pass.get();
+            string name = typeid(*p).name();
```
Comment (Krovatkin) [inline src/ngraph/pass/manager.cpp]: do you think we could get rid of `m_pass_names` for good and just use this logic to get pass names?

File: src/ngraph/pass/manager.cpp [hunk 3]
```diff
@@ -122,6 +131,19 @@ void ngraph::pass::Manager::run_passes(shared_ptr<Function> func)
             }
         }
         index++;
+        pass_timer.stop();
+        if (profile_enabled)
```
Comment (Krovatkin) [inline src/ngraph/pass/manager.cpp]: this is a bit inconsistent with `m_visualize` and `m_serialize`. I don't know if we want consistency here like `m_profile` which is initialized in `PassManager::PassManager`?

File: src/ngraph/pass/manager.cpp [hunk 4]
```diff
@@ -55,6 +58,8 @@ void ngraph::pass::Manager::initialize_default_passes()
 
 void ngraph::pass::Manager::run_passes(shared_ptr<Function> func)
 {
+    bool profile_enabled = getenv("NGRAPH_PROFILE_PASS_ENABLE") != nullptr;
```
Comment (Krovatkin) [inline src/ngraph/pass/manager.cpp]: we could probably make it `static const` to signify that this value won't change?

File: src/ngraph/pass/manager.cpp [hunk 5]
```diff
@@ -15,6 +15,8 @@
 *******************************************************************************/
 
 #include <algorithm>
+#include <cxxabi.h>
```
Comment (Krovatkin) [inline src/ngraph/pass/manager.cpp]: I wonder if this will work on macosx, let alone windows?

File: src/ngraph/pass/manager.cpp (hunk 1 above)
Comment (rkimballn1) [reply src/ngraph/pass/manager.cpp]: My goal is to blow your mind

File: src/ngraph/pass/manager.cpp (hunk 2 above)
Comment (rkimballn1) [reply src/ngraph/pass/manager.cpp]: yes

File: src/ngraph/pass/manager.cpp (hunk 4 above)
Comment (rkimballn1) [reply src/ngraph/pass/manager.cpp]: const is ok, static is meh

File: src/ngraph/pass/manager.cpp (hunk 2 above)
Comment (rkimballn1) [reply src/ngraph/pass/manager.cpp]: No, too big a change. Needs a separate PR.

File: src/ngraph/pass/manager.cpp (hunk 3 above)
Comment (rkimballn1) [reply src/ngraph/pass/manager.cpp]: This is only used in this one method.

**Pull request merged**